### PR TITLE
Correct logging with exceptions

### PR DESCRIPTION
**Pull request by rorygraves**
Address exception logging issues from #1526

### CODE REVIEW

**Comment by fommil** [comment]
LGTM

File: core/src/main/scala/org/ensime/indexer/SearchService.scala [hunk 1]
```diff
@@ -332,23 +332,23 @@ class IndexingQueueActor(searchService: SearchService) extends Actor with ActorL
...
+          log.error(t, s"failed to index batch of ${batch.size} files")
         case Success(indexed) =>
           searchService.delete(indexed.map(_._1)(collection.breakOut)).onComplete {
             case Failure(t) =>
               searchService.semaphore.release()
-              log.error(s"failed to remove stale entries in ${batch.size} files", t)
+              log.error(t, s"failed to remove stale entries in ${batch.size} files")
             case Success(_) => indexed.foreach {
               case (file, syms) =>
-                val boost = searchService.isUserFile((file.getName))
```
Comment (fommil) [inline core/src/main/scala/org/ensime/indexer/SearchService.scala]: this double parenthesis is needed for the presentation compiler to understand it

File: core/src/main/scala/org/ensime/indexer/SearchService.scala [hunk 2]
```diff
@@ -135,7 +135,7 @@ class SearchService(
     // index all the given bases and return number of rows written
     def indexBases(bases: Set[FileObject], checks: Seq[FileCheck]): Future[Int] = {
       log.debug("Indexing bases...")
-      val checksLookup: Map[String, FileCheck] = checks.map(check => (check.filename -> check)).toMap
```
Comment (fommil) [inline core/src/main/scala/org/ensime/indexer/SearchService.scala]: you might want to back out some of these cleanups as it'll introduce conflicts with the graphpocalypse branch.

**Comment by rorygraves** [comment]
Will back out everything but the logging change shortly.

**Comment by fommil** [comment]
LGTM

File: core/src/main/scala/org/ensime/indexer/SearchService.scala (hunk 1 above)
Comment (fommil) [inline core/src/main/scala/org/ensime/indexer/SearchService.scala]: (we should file a bug for this with the presentation compiler)

**Comment by fommil** [comment]
LGTM

**Comment by rorygraves** [comment]
will port to 2.0 later

**Pull request merged**